camera_calibration/src/camera_calibration/camera_calibrator.py

```diff
@@ -387,3 +451,43 @@ def redraw_stereo(self, drawable):
 
         self._last_display = display
         self.queue_display.put(display)
+
+    def ask_for_calibration(self):
+        if self.c.goodenough:
+            do_calibrate = ask_yes_no_question("Current collected set of samples "
+                                               "is good enough, do you want to calibrate?")
+            if do_calibrate:
+                print("**** Calibrating ****")
+                rms = self.c.do_calibration()
+                if self.c.calibrated:
+                    print("RMS Error from this calibration is: {}".format(str(rms)))
+                    do_save = ask_yes_no_question("Do you want to save this result and exit? "
+                                                  "If no then collecting new samples.")
+                    if do_save:
+                        self.c.do_save()
+                        rospy.signal_shutdown("Quit")
+            self.c.calibrated = False
+            print("Continuing with collecting new samples...")
+
+
+def ask_yes_no_question(question, default_yes=True):
+    """
+    Ask the user a question via a command line prompt.
+    The user should answer, y, yes, n, no or some upper-/lowercase
+    variation thereof, with empty input interpreted as the default
+    answer (yes if not specified when calling this method)
+    :param default_yes: bool with the default answer
+    :param question: string with the question to be asked
+    """
+    while True:
+        answer = input(
+            question + (" [Y/n]: " if default_yes else " [y/N]: ")).lower()
+        if answer in ["y", "ye", "yes"]:
+            return True
+        elif answer in ["n", "no"]:
+            return False
+        elif answer == "":
+            return default_yes
+        else:
+            print("Please answer using only y or n (or yes/no if you prefer)")
```

Comment:
the else is not necessary here. And we have trailing newlines at the end of the file